lib/compilers/coccinelle.ts

```diff
@@ -0,0 +1,281 @@
+import {SelectedLibraryVersion} from '../../types/libraries/libraries.interfaces.js';
+import {BaseCompiler} from '../base-compiler.js';
+import {CompilationEnvironment} from '../compilation-env.js';
+import {logger} from '../logger.js';
+import * as utils from '../utils.js';
+
+export class CoccinelleCCompiler extends BaseCompiler {
+    protected spatchBaseFilename: string; // if true, doTempfolderCleanup won't clean up
+    protected joinSpatchStdinAndStderr: boolean; // dirty hopefullytemporary hack, as coccinelle dumps diagnostics on both streams
+
+    static get key() {
+        return 'coccinelle_for_c';
+    }
+
+    constructor(info: PreliminaryCompilerInfo, env: CompilationEnvironment) {
+        super(
+            {
+                disabledFilters: ['labels', 'directives', 'commentOnly', 'trim', 'debugCalls'],
+                ...info,
+            },
+            env,
+        );
+        this.compiler.supportsIntel = true;
+        this.delayCleanupTemp = false;
+        this.spatchBaseFilename = 'patch.cocci';
+        this.outputFilebase = 'output';
+        this.joinSpatchStdinAndStderr = true;
+        // ...
+    }
+
+    override getDefaultFilters() {
+        // We want to keep the entire output of Coccinelle (so, no filter)
+        return {
+            intel: false,
+            commentOnly: false,
+            directives: false,
+            labels: false,
+            optOutput: false,
+            binary: false,
+            execute: false,
+            demangle: false,
+            libraryCode: false,
+            trim: false,
+            binaryObject: false,
+            debugCalls: false,
+        };
+    }
+
+    override async populatePossibleOverrides() {
+        await super.populatePossibleOverrides();
+    }
+
+    override getSharedLibraryPathsAsArguments(libraries: CompileChildLibraries[], libDownloadPath?: string) {
+        return [];
+    }
+
+    override getSharedLibraryLinks(libraries: any[]): string[] {
+        return [];
+    }
+
+    override getIncludeArguments(libraries: SelectedLibraryVersion[], dirPath: string): string[] {
+        return super.getIncludeArguments(libraries, dirPath);
+    }
+
+    override orderArguments(
+        options: string[],
+        inputFilename: string,
+        libIncludes: string[],
+        libOptions: string[],
+        libPaths: string[],
+        libLinks: string[],
+        userOptions: string[],
+        staticLibLinks: string[],
+    ) {
+        return options.concat(userOptions, libIncludes, libOptions, libPaths, libLinks, staticLibLinks, [
+            this.filename(inputFilename),
+        ]);
+    }
+
+    override optionsForBackend(backendOptions: Record<string, any>, outputFilename: string) {
+        return super.optionsForBackend(backendOptions, outputFilename);
+    }
+
+    override optionsForFilter(filters: ParseFiltersAndOutputOptions, outputFilename: string, userOptions?: string[]) {
+        // coccinelle_for_c
```

Comment:
what does this comment refer to?

Reply:
I introduced two "languages": `coccinelle_for_c` and `coccinelle_for_cpp`.
The source code window on the left offers therefore two "languages": "C with Coccinelle" and "C++ with Coccinelle".
The class handling `coccinelle_for_cpp` is `CoccinelleCPlusPlusCompiler` and extends `CoccinelleCompiler`; as of now these are both in `coccinelle.ts`.